supabase/functions/chat/index.ts

```diff
@@ -21,29 +40,48 @@ const chat = async (req) => {
     throw new ApplicationError(
       "Unable to get auth user details in request data"
     );
-  const { messageHistory } = await req.json();
+  const requestBody = await req.json();
+  const { messageHistory, userMessage } = requestBody;
+
+  // Append userMessage to messageHistory if it exists
+  if (userMessage) {
```

Comment:
it will never exist, as the client just send the whole `messageHistory` as is.  I guess we can remove it?

`const { messageHistory } =`